fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/ElectInstance.java

```diff
@@ -43,17 +43,12 @@ public ElectInstance(
   }
 
   public ElectInstance(ElectInstance electInstance) {
-    this.messageId = new MessageID(electInstance.messageId);
-    this.channel = new Channel(electInstance.channel);
-    this.proposer = new PublicKey(electInstance.proposer);
-    this.elect = new ConsensusElect(electInstance.elect);
-    this.nodes = electInstance.nodes.stream().map(PublicKey::new).collect(Collectors.toSet());
-    this.acceptorToMessageId =
-        electInstance.acceptorToMessageId.entrySet().stream()
-            .collect(
-                Collectors.toMap(
-                    entry -> new PublicKey(entry.getKey()),
-                    entry -> new MessageID(entry.getValue())));
+    this.messageId = electInstance.messageId;
+    this.channel = electInstance.channel;
+    this.proposer = electInstance.proposer;
+    this.elect = electInstance.elect;
+    this.nodes = electInstance.nodes;
```

Comment:
don't we need to copy the `ConsensusElect` ? 
and if not, can we annotate it so that future changes would have to explicitly break our assumptions ?

Reply:
ConsensusElect is part of the network data classes.

Our current assumption (that I documented a second ago 😄) is that classes representing messages are immutable.